integration-core/src/main/java/org/symphonyoss/integration/core/bootstrap/IntegrationBootstrapContext.java

```diff
@@ -129,11 +138,31 @@ public void initIntegrations() {
 
       // deals with unknown apps.
       initUnknownApps();
+
+      // Start health check polling
+      healthCheckAgentServicePolling();
     }
 
   }
 
   /**
+   * Schedule to dispatch health-check service event to monitor the Agent version.
+   */
+  private void healthCheckAgentServicePolling() {
+    scheduler.scheduleWithFixedDelay(new Runnable() {
+
+      @Override
+      public void run() {
+        LOGGER.debug("Polling AGENT health check");
+
+        HealthCheckServiceEvent event = new HealthCheckServiceEvent(AGENT_SERVICE_NAME);
+        publisher.publishEvent(event);
```

Comment:
This seems like a lot of indirection.  You've got one thread that sends events.  A listener receives that event and calls Agent health check. Based on results of health check, the listener may fire another event that says version of Agent has changed.  Listener for version changed event then stores the new value.  Why isn't the health check call just it's own recurring runnable?